library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/utils/WireframeExtTest.kt

```diff
@@ -0,0 +1,178 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.utils
+
+import com.datadog.android.sessionreplay.forge.ForgeConfigurator
+import com.datadog.android.sessionreplay.model.MobileSegment
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.annotation.FloatForgery
+import fr.xgouchet.elmyr.annotation.Forgery
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+import java.util.stream.Stream
+
+@Extensions(ExtendWith(ForgeExtension::class))
+@ForgeConfiguration(ForgeConfigurator::class)
+internal class WireframeExtTest {
+
+    // region hasOpaqueBackground
+
+    @ParameterizedTest
+    @MethodSource("testWireframes")
+    fun `M return true W hashOpaqueBackground { shapeStyle fully opaque with color }`(
+        fakeWireframe: MobileSegment.Wireframe,
+        @StringForgery(regex = "#[0-9A-Fa-f]{6}[fF]{2}") fakeOpaqueColor: String,
+        forge: Forge
+    ) {
+        // Given
+        val fakeShapeStyle: MobileSegment.ShapeStyle = forge.getForgery<MobileSegment.ShapeStyle>()
+            .copy(
+                backgroundColor = fakeOpaqueColor,
+                opacity = forge.anElementFrom(listOf(1f, 1))
```

Comment:
why do we have 1 as float and as integer here? should it be 2 different values?

Reply:
is just that the `ShapeStyle` uses `Number` so I wanted to check with int and float values also. Normally we only use `float` for that value in the code but just in case.